### PR TITLE
UPP MMG

### PR DESCRIPTION
**Pull request by Steelpoint**
# About the pull request

Adds in one feature from this now auto-closed PR: https://github.com/cmss13-devs/cmss13/pull/5945

Adds the T37 MMG for the UPP, their alternative version to the USCM M2C. The T37 is a stationary weapon, meaning it can not be moved, it has a slower rate-of-fire but it can sustain firing for significantly longer before it will overheat. 

# Explain why it's good for the game

Useful features for mapping and events. 

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: Adds the T37 Medium Machine Gun to the UPP's arsenal. The T37 is an alternative variant of the M2C HMG, it is a stationary emplacement (that can not be redeployed) that has a slower rate of fire compared to the M2C, but it can fire for a far longer period of time before overheating. This is a purely defensive weapon, and it must be pre-placed on a map by a mapper. 
/:cl:

### CODE REVIEW

**Review by Blundir — CHANGES_REQUESTED** [review]
The gun and the ammo box is literally slight M2C tweak, find or make an actual sprite for both, we have special sprites for upp ammo boxes and special palette for upp guns.

**Comment by Steelpoint** [comment]
Moving the UPP Map to a separate PR.

**Comment by Waseemq1235** [comment]
Just a heads-up: the MMG values I used in PR #5945 were tested but not extensively.

**Comment by Steelpoint** [comment]
> Just a heads-up: the MMG values I used in PR #5945 were tested but not extensively.

We'll likely get some chances to see it in action if the PR gets merged so a future balance PR can be done if its excessive.

As it stands, it trades huge burst damage for sustained suppressing fire, plus the lack of being able to move the weapon, and the fact you'll only see these guns being placed by mappers and admins, not players, makes it easier to balance and less oppressive.

**Comment by Steelpoint** [comment]
Love it, thank you.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/modules/cm_marines/m2c.dm [hunk 1]
```diff
@@ -589,6 +606,95 @@
...
+	update_icon() //final safeguard.
+	var/obj/item/ammo_magazine/m2c/t37/AM = new /obj/item/ammo_magazine/m2c/t37(src.loc)
+	AM.current_rounds = 0
+	AM.update_icon()
+
+/obj/structure/machinery/m56d_hmg/auto/t37/attackby(obj/item/O as obj, mob/user as mob)
+	if(!ishuman(user))
+		return
+	// RELOADING
+	if(istype(O, /obj/item/ammo_magazine/m2c/t37))
```
Comment (harryob) [inline code/modules/cm_marines/m2c.dm:655]: are we overriding this whole proc just to change this line? ideally we would change the parent proc in /obj/structure/machinery/m56d_hmg/auto like:

```dm

if(istype(O, /obj/item/ammo_magazine)
	var/obj/item/ammo_magazine/magazine = O
	
	if(magazine.default_ammo != ammo)
		return
```

File: code/modules/cm_marines/m2c.dm [hunk 2]
```diff
@@ -589,6 +606,95 @@
...
+		qdel(src)
+		return
+	else
+		..()
+
+/obj/structure/machinery/m56d_hmg/auto/t37/handle_ammo_out(mob/user)
+	visible_message(SPAN_NOTICE("[icon2html(src, viewers(src))] [src]'s ammo box drops onto the ground, now completely empty."))
+	playsound(loc, empty_alarm, 70, 1)
+	update_icon() //final safeguard.
+	var/obj/item/ammo_magazine/m2c/t37/AM = new /obj/item/ammo_magazine/m2c/t37(src.loc)
```
Comment (harryob) [inline code/modules/cm_marines/m2c.dm:647]:
```suggestion
	var/obj/item/ammo_magazine/m2c/t37/AM = new(loc)
```

File: code/modules/cm_marines/m2c.dm [hunk 3]
```diff
@@ -589,6 +606,95 @@
...
+	if(health <= 0)
+		if(prob(explosion_chance))
+			var/cause_data = create_cause_data(src.name)
+			visible_message(SPAN_HIGHDANGER("[src] explodes into pieces!"))
+			cell_explosion(loc, explosion_power, explosion_falloff, falloff_mode, null, cause_data)
+		playsound(src.loc, 'sound/items/Welder2.ogg', 25, 1)
+		qdel(src)
+		return
+	else
+		..()
```
Comment (harryob) [inline code/modules/cm_marines/m2c.dm:641]:
```suggestion
	return ..()
```
previous condition returns so you don't need the else

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by Steelpoint** [comment]
I'll fix this up (later).

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/modules/cm_marines/m2c.dm [hunk 4]
```diff
@@ -330,22 +343,25 @@
...
+            to_chat(user, SPAN_WARNING("There's already an ammo box inside of [src], remove it first!"))
+            return
+        if(user.action_busy) return
+        user.visible_message(SPAN_NOTICE("[user] loads [src] with an ammo box! "), SPAN_NOTICE("You load [src] with an ammo box!"))
+        playsound(src.loc, 'sound/items/m56dauto_load.ogg', 75, 1)
+        rounds = min(rounds + M.current_rounds, rounds_max)
+        update_icon()
+        user.temp_drop_inv_item(O)
+        qdel(O)
+        return
```
[the remark this answers is not in the document]
Comment (cuberound) [reply code/modules/cm_marines/m2c.dm:364]: dd

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by harryob — APPROVED** [review]

**Pull request merged**